### PR TITLE
password hashing is done

### PR DESCRIPTION
**Pull request by andromash**

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
Well Done!

File: src/main/java/com/internet/shop/model/User.java [hunk 1]
```diff
@@ -23,6 +24,21 @@ public User(Long id, String name, String login, String password, Set<Role> roles
...
     }
 
+    public User(String name, String login, String password, byte[] salt) {
+        this.name = name;
+        this.login = login;
+        this.password = password;
+        this.salt = salt;
+    }
+
+    public User(Long id, String name, String login, String password, byte[] salt) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/User.java]: I am conserned about so many constructors in this class) this is not a good practice and should better be refactored

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -15,7 +16,7 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
-        if (userFromDB.isEmpty() || !userFromDB.get().getPassword().equals(password)) {
+        if (userFromDB.isEmpty() || !HashUtil.isValid(password, userFromDB.get())) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: let's move the private method to this class

we pass user to the method
HashUtil should not have methods with such a specific business logic

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/model/User.java (hunk 1 above)
Comment (andromash) [reply src/main/java/com/internet/shop/model/User.java]: agree, they were unused, already deleted them

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -16,9 +16,14 @@
...
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
-        if (userFromDB.isEmpty() || !HashUtil.isValid(password, userFromDB.get())) {
+        if (userFromDB.isEmpty() || isNotValid(password, userFromDB.get())) {
             throw new AuthenticationException("Incorrect login or password");
         }
         return userFromDB.get();
     }
+
+    private static boolean isNotValid(String password, User userFromDB) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]:
```suggestion
    private boolean isNotValid(String password, User userFromDB) {
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**